File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int profit = 0;
        int temp = prices[0];
        for(int i = 1; i < n; i++) {
            if(prices[i] < temp) {
                temp = prices[i];
            }
            else if(prices[i] - temp > profit) {
                profit = prices[i] - temp;
            }
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:

        n = len(nums)
        minV = nums[0]

        profit = 0

        for i in range(1, n):

            minV = min(minV, nums[i])

            profit = max (profit, nums[i] - minV)

        
        return profit